### PR TITLE
[ConsumerEvent] Add metadata to data payload

### PR DESCRIPTION
**Pull request by acco**

### CODE REVIEW

**Comment by acco** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/111?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#111** <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/111?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#110** <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/110?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @acco and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: lib/sequin/consumers/consumer_event.ex [hunk 1]
```diff
@@ -12,18 +12,32 @@ defmodule Sequin.Consumers.ConsumerEvent do
...
       field :record, :map
       field :changes, :map
       field :action, Ecto.Enum, values: [:insert, :update, :delete]
+
+      embeds_one :metadata, Metadata, primary_key: false do
+        @derive Jason.Encoder
+        field :table, :string
+        field :schema, :string
+        field :commit_timestamp, :utc_datetime_usec
+      end
```
[the remark this answers is not in the document]
Comment (RTLS) [reply lib/sequin/consumers/consumer_event.ex:27]: It was helpful last time!

File: lib/sequin/replication/message_handler.ex [hunk 2]
```diff
@@ -58,15 +57,25 @@ defmodule Sequin.Replication.MessageHandler do
...
-      action: :delete
+      action: :delete,
+      metadata: metadata(message)
+    }
+  end
+
+  defp metadata(%Message{} = message) do
+    %{
+      table: message.table_name,
+      schema: message.table_schema,
```
Comment (RTLS) [inline lib/sequin/replication/message_handler.ex:77]: just curious why the name changes (table_ prefix gets dropped)

**Review by RTLS — APPROVED** [review]

File: lib/sequin/replication/message_handler.ex (hunk 2 above)
Comment (acco) [reply lib/sequin/replication/message_handler.ex:77]: Hmm... because the whole map isn't just about the table, we should probably preserve.

**Pull request merged**